src/notation/internal/notationnoteinput.cpp

```diff
@@ -255,6 +258,65 @@ void NotationNoteInput::putTuplet(const TupletOptions& options)
     m_stateChanged.notify();
 }
 
+QRectF NotationNoteInput::cursorRect() const
+{
+    if (!isNoteInputMode()) {
+        return QRectF();
+    }
+
+    Ms::InputState& inputState = score()->inputState();
+    Ms::Segment* segment = inputState.segment();
+    if (!segment) {
+        return QRectF();
+    }
+
+    Ms::System* system = segment->measure()->system();
+    if (!system) {
+        return QRectF();
+    }
+
+    int track = inputState.track() == -1 ? 0 : inputState.track();
+    int staffIdx = track / VOICES;
+
+    static constexpr int sideMargin = 4;
+    static constexpr int skylineMargin = 20;
+
+    QRectF segmentContentRect = segment->contentRect();
+    double x = segmentContentRect.translated(segment->pagePos()).x() - sideMargin;
+    double y = system->staffYpage(staffIdx) + system->page()->pos().y();
+    double w = segmentContentRect.width() + 2 * sideMargin;
+    double h = 0.0;
+
+    Staff* staff = score()->staff(staffIdx);
+    const Ms::StaffType* staffType = staff->staffType(inputState.tick());
+    double spatium = score()->spatium();
+    double lineDist = staffType->lineDistance().val() * spatium;
+    int lines = staffType->lines();
+    int strg = inputState.string();
+
+    int instrStrgs = staff->part()->instrument()->stringData()->strings();
+    if (staff->isTabStaff(inputState.tick()) && strg >= 0 && strg <= instrStrgs) {
+        h = lineDist;
+        y += staffType->physStringToYOffset(strg) * spatium;
+        y -= (staffType->onLines() ? lineDist * 0.5 : lineDist);
+    } else {
+        h = (lines - 1) * lineDist + 2 * skylineMargin;
+        y -= skylineMargin;
+    }
+
+    QRectF result = QRectF(x, y, w, h).translated(system->page()->pos());
+    return result;
+}
+
+QColor NotationNoteInput::cursorColor() const
```

Comment:
Do we really need this method? There is currentVoiceIndex in NoteInputState already. Seems that you can use it for resolving selection color